.github/workflows/osrm-backend-docker.yml

```diff
@@ -0,0 +1,84 @@
+name: build and publish container image
+on:
+  push:
+    tags:
+      - 'v*'
+    branches:
+      - master      
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:       
+      - name: Check out the repo
+        uses: actions/checkout@v2
+
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: ghcr.io/${{ github.repository }}/osrm-backend
+
+      - name: Docker meta - debug
+        id: metadebug
+        uses: docker/metadata-action@v3
+        with:
+          images: ghcr.io/${{ github.repository }}/osrm-backend          
+          flavor: |
+            latest=true
+            suffix=-debug,onlatest=true
+
+      - name: Docker meta - assertions
+        id: metaassertions
+        uses: docker/metadata-action@v3
+        with:
+          images: ghcr.io/${{ github.repository }}/osrm-backend          
+          flavor: |
+            latest=true
+            suffix=-assertions,onlatest=true
+          
+
+
+
+      - name: Log in to GitHub Docker Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+
+
+
+      - name: Build container image - normal
+        uses: docker/build-push-action@v2
+        with:
+          push: true
+          file: ./docker/Dockerfile
+          tags:  ${{ steps.meta.outputs.labels }}
+          build-args: 
+          - DOCKER_TAG:  ${{ steps.meta.outputs.labels }}
```

Comment:
Looks like `build-args` expects a list, not a dictionary.
>  build-args | List | List of build-time variables

With the `github.repository` change, this file should be fork agnostic, you can try building the containers on your own fork to confirm it works.